x-pack/filebeat/module/gcp/cloud_armor/config/pipeline.js

```diff
@@ -0,0 +1,244 @@
+        field: "json.timestamp",
+        timezone: "UTC",
+        layouts: ["2006-01-02T15:04:05.999999999Z07:00"],
+        tests: ["2019-06-14T03:50:10.845445834Z"],
+        ignore_missing: true,
+    });
+
+    var saveOriginalMessage = function(evt) {};
+    if (keep_original_message) {
+        saveOriginalMessage = new processor.Convert({
+            fields: [
+                {from: "message", to: "event.original"}
+            ],
+            mode: "rename"
+        });
+    }
+
+    var dropPubSubFields = function(evt) {
+        evt.Delete("message");
+    };
+
+    var saveMetadata = new processor.Convert({
+        fields: [
+            {from: "json.logName", to: "log.logger"},
+            {from: "json.insertId", to: "event.id"},
+        ],
+        ignore_missing: true
+    });
+
+    // Use the monitored resource type's labels to set the cloud metadata.
+    // The labels can vary based on the resource.type.
+    var setCloudMetadata = new processor.Convert({
+        fields: [
+            {
+                from: "json.resource.labels.project_id",
+                to: "cloud.project.id",
+                type: "string"
+            },
+            {
+                from: "json.resource.labels.backend_service_name",
+                to: "cloud.backend.name",
+                type: "string"
+            }
+        ],
+        ignore_missing: true,
+        fail_on_error: false,
+    });
+
+    // Keep httpRequest metadata in message
+    var convertHttpRequest = new processor.Convert({
+        fields: [
+            {
+                from: "json.httpRequest.requestMethod",
+                to: "gcp.cloud_armor.http_request.method",
+                type: "string"
+            },
+            {
+                from: "json.httpRequest.requestUrl",
+                to: "gcp.cloud_armor.http_request.url",
+                type: "string"
+            },
+            {
+                from: "json.httpRequest.requestSize",
+                to: "gcp.cloud_armor.http_request.request_size",
+                type: "string"
+            },
+            {
+                from: "json.httpRequest.status",
+                to: "gcp.cloud_armor.http_request.status_code",
+                type: "integer"
+            },
+            {
+                from: "json.httpRequest.responseSize",
+                to: "gcp.cloud_armor.http_request.response_size",
+                type: "string"
+            },
+            {
+                from: "json.httpRequest.userAgent",
+                to: "gcp.cloud_armor.http_request.user_agent",
+                type: "string"
+            },
+            {
+                from: "json.httpRequest.remoteIp",
+                to: "gcp.cloud_armor.http_request.remote_ip",
+                type: "ip"
+            },
+            {
+                from: "json.httpRequest.serverIp",
+                to: "gcp.cloud_armor.http_request.server_ip",
+                type: "ip"
+            },
```

Comment:
Do you think it's worth having Referrer in this list? So we can attempt to see (where possible) the path taken? Not sure if all that much value from an attack side would be gained so open to leaving this one

Reply:
Very true, weirdly i didn't have a referer in my log example when writing this! But yes it should be there: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest

Reply:
Added in https://github.com/unravelin/beats/pull/1/commits/d2333c4d8da374ec528bc59cc3f8f5a087d647bc